fv3core/fv3core/initialization/baroclinic.py

```diff
@@ -0,0 +1,522 @@
+    if ptop < ptop_min:
+        ak1 = (constants.KAPPA + 1.0) / constants.KAPPA
+        peln[:, :, 0] = peln[:, :, 1] - ak1
+    else:
+        peln[:, :, 0] = np.log(ptop)
+
+
+def p_var(
+    delp,
+    delz,
+    pt,
+    ps,
+    qvapor,
+    pe,
+    peln,
+    pkz,
+    ptop,
+    moist_phys,
+    make_nh,
+    hydrostatic=False,
+    adjust_dry_mass=False,
+):
+    """
+    Computes auxiliary pressure variables for a hydrostatic state.
+
+    The Fortran code also recomputes some more pressure variables,
+    pe, pk, but since these are already done in setup_pressure_fields
+    we don't duplicate them here
+    """
+    assert not adjust_dry_mass
+    assert not hydrostatic
+
+    ps[:] = pe[:, :, -1]
+    fix_top_log_edge_pressure(peln, ptop)
+
+    if not hydrostatic:
+        if make_nh:
+            delz[:, :, :-1] = initialize_delz(pt, peln)
+        if moist_phys:
+            pkz[:, :, :-1] = initialize_pkz_moist(delp, pt, qvapor, delz)
+        else:
+            pkz[:, :, :-1] = initialize_pkz_dry(delp, pt, delz)
+
+
+# TODO: maybe extract from quantity related objects
+def local_compute_size(data_array_shape):
+    nx = data_array_shape[0] - 2 * nhalo - 1
+    ny = data_array_shape[1] - 2 * nhalo - 1
+    nz = data_array_shape[2]
+    return nx, ny, nz
+
+
+def compute_slices(nx, ny):
+    islice = slice(nhalo, nhalo + nx)
+    jslice = slice(nhalo, nhalo + ny)
+    slice_3d = (islice, jslice, slice(None))
+    slice_2d = (islice, jslice)
+    return islice, jslice, slice_3d, slice_2d
+
+
+def init_baroclinic_state(
+    metric_terms: MetricTerms,
+    adiabatic: bool,
+    hydrostatic: bool,
+    moist_phys: bool,
+    comm: fv3util.CubedSphereCommunicator,
+):
+    """
+    Create a DycoreState object with quantities initialized to the Jablonowski &
+    Williamson baroclinic test case perturbation applied to the cubed sphere grid.
+    """
+
+    state = DycoreState.init_empty(metric_terms.quantity_factory)
+    nx, ny, nz = local_compute_size(state.delp.data.shape)
+    # Initializing to values the Fortran does for easy comparison
+    state.delp.data[:] = 1e30
+    state.delp.data[:nhalo, :nhalo] = 0.0
+    state.delp.data[:nhalo, nhalo + ny :] = 0.0
+    state.delp.data[nhalo + nx :, :nhalo] = 0.0
+    state.delp.data[nhalo + nx :, nhalo + ny :] = 0.0
+    state.pe.data[:] = 0.0
+    state.pt.data[:] = 1.0
+    state.ua.data[:] = 1e35
+    state.va.data[:] = 1e35
+    state.uc.data[:] = 1e30
+    state.vc.data[:] = 1e30
+    state.w.data[:] = 1.0e30
+    state.delz.data[:] = 1.0e25
+    state.phis.data[:] = 1.0e25
+    state.ps.data[:] = jablo_init.surface_pressure
+    eta = np.zeros(nz)
+    eta_v = np.zeros(nz)
+    islice, jslice, slice_3d, slice_2d = compute_slices(nx, ny)
+    # Slices with extra buffer points in the horizontal dimension
+    # to accomodate averaging over shifted calculations on the grid
+    isliceb, jsliceb, slice_3db, slice_2db = compute_slices(nx + 1, ny + 1)
```

Comment:
I get that b means b-grid which should mean interfaces (or maybe it means buffer?), but this is one of the harder to understand naming conventions for grad students - is there any way to avoid this?

Two of the outputs appear to be unused:
```suggestion
    _, _, slice_3db, slice_2db = compute_slices(nx + 1, ny + 1)
```